Applying option to enable/disable map move on coordinate update

murraypurves opened the pull request:
Reapplying the changes to the new codebase as requested.

murraypurves committed:
Applying option to enable/disable map move on coordinate update

igaurab commented:
Hi, My device doesn't show up the marker when the updateMapOnPositionChange is set to false. The same issue was also raised by @jesusrp98 in #3.
I have created an issue #7, please have a look at it

murraypurves committed:
Update user_location_options.dart

Refactoring: changing name of updateMapLocationOnPositionChange

murraypurves committed:
Update user_location_layer.dart

Refactoring: rename updateMapLocationOnPositionChange

murraypurves committed:
Update main.dart

Refactoring: rename updateMapLocationOnPositionChange

murraypurves committed:
Update README.md

Refactoring: rename updateMapLocationOnPositionChange

murraypurves commented:
Are you manually moving the map to where you know your GPS position to be? The flag only controls the bit where the map is moved, it shouldn't affect the appearance of the marker. 

Try manually moving the map to contain where you know your position to be (for example, in your test program I would have to zoom out & move the map to the UK, but I can see it when I do).

I have refactored my branch to rename the variable to something a bit more descriptive; apologies, I have done it directly on GitHub so it had to be one commit per file...

W.r.t. the mapController, if it is only needed to move the map then it could potentially be removed as part of my proposed update. I can have a look at this tomorrow night - I think we can use the `required` keyword in the constructor and a little bit of logic to disable the move if a controller is not provided.

murraypurves committed:
Making mapController optional.
Updating constructor to specify required parameters, adding checks to allow mapController to be null

murraypurves commented:
I have tested again, and I can't replicate your issue - I can see my GPS marker when `updateMapLocationOnPositionChange` is set to false, I just have to manually move the map over.

The latest commit makes the `mapController` optional (per #7); you should be able to use the plugin without it now (though the map position update will be forced disabled). I have also marked some of the parameters in the constructor as required to ensure that the minimum requirements are met when it is instantiated.

igaurab commented:
@murraypurves thanks, the `mapController` worked as expected this time.

igaurab commented on lib/src/user_location_options.dart:
```diff
@@ -11,15 +11,15 @@ class UserLocationOptions extends LayerOptions {
-    this.markers,
+    @required this.context,
+    @required this.markers,
+    @required this.markerlocationStream,
```
`markerLocationStream`  provides the current location of the marker, this functionality might not be required in all use cases. Let's make this optional parameter
